File: Apps/Codaxy.Dextop.Showcase/Demos/Remoting/RemoteInstantiation.js
Ext.define('Showcase.demos.RemoteInstantiationWindow', {
	extend: 'Dextop.Window',
	width: 300,
	height: 250,
	title: 'Dextop Remote Instantiation',

	initComponent: function () {

		Ext.apply(this, {
			bodyStyle: 'padding: 5px',
			layout: {
				type: 'vbox',
				align: 'stretch'
			},
			defaults: {
				xtype: 'button',
				margins: '3 3 3 3',
				flex: 1,
				scope: this
			},
			items: [{
				text: 'Parameterless',
				handler: function () {
					Dextop.getSession().remote.Instantiate('Showcase.demos.RemoteInstantiationWindow', null, {
						type: 'alert',
						scope: this,
						success: function (config) {
							var win = Dextop.create(config, {
								modal: true,
								x: this.x + 50,
								y: this.y + 50
							});
							win.show();
						}
					});
				}
			}, {
				text: 'Invoke using an alias',
				handler: function () {
					Dextop.getSession().remote.Instantiate('rinstant', null, {
						type: 'alert',
						scope: this,
						success: function (config) {
							var win = Dextop.create(config, {
								modal: true,
								x: this.x + 50,
								y: this.y + 50
							});
							win.show();
						}
					});
				}
			}, {
				text: 'Specify params as an array',
				handler: function () {
					Dextop.getSession().remote.Instantiate('Showcase.demos.RemoteInstantiationWindow', [1, 2], {
						type: 'alert',
						scope: this,
						success: function (config) {
							var win = Dextop.create(config, {
								modal: true,
								x: this.x + 50,
								y: this.y + 50
							});
							win.show();
						}
					});
				}
			}, {
				text: 'Specify params as a hash',
				handler: function () {
					Dextop.getSession().remote.Instantiate('Showcase.demos.RemoteInstantiationWindow', { a: 1, b: 2 }, {
						type: 'alert',
						scope: this,
						success: function (config) {
							var win = Dextop.create(config, {
								modal: true,
								x: this.x + 50,
								y: this.y + 50
							});
							win.show();
						}
					});
				}
			}, {
				text: 'Send a hash',
				handler: function () {
					Dextop.getSession().remote.Instantiate('Showcase.demos.RemoteInstantiationWindow', { msg: 'Everything is allowed' }, {
						type: 'alert',
						scope: this,
						success: function (config) {
							var win = Dextop.create(config, {
								modal: true,
								x: this.x + 50,
								y: this.y + 50
							});
							win.show();
						}
					});
				}
			}, {
			    text: 'Using a constructor route',
			    handler: function () {
			        Dextop.getSession().remote.Instantiate('rinstant/param', null, {
			            type: 'alert',
			            scope: this,
			            success: function (config) {
			                var win = Dextop.create(config, {
			                    modal: true,
			                    x: this.x + 50,
			                    y: this.y + 50
			                });
			                win.show();
			            }
			        });
			    }
			}]
		});
		this.callParent(arguments);
	}
});

File: Apps/Codaxy.Dextop.Showcase/Demos/Remoting/RemoteInstantiation.cs
using System;
using System.Threading;
using Codaxy.Dextop.Remoting;

namespace Codaxy.Dextop.Showcase.Demos.Remoting
{
    [Demo("RemoteInstantiation",
        Title = "Remote Instantiation",
        Description = "Learn how to create remote objects on the server.",
        Path = "~/Demos/Remoting"
    )]
    [LevelMedium]
    [TopicDextopRemoting]
	[CategoryGettingStarted]
    public class RemoteInstantiationWindow : DextopWindow
    {
        String title;
        
        [DextopRemotableConstructor(alias = "rinstant")]
        public RemoteInstantiationWindow()
        {
            title = "Parameterless";
        }

        [DextopRemotable]
        public RemoteInstantiationWindow(String a, String b)
        {
            title = String.Format("[{0} {1}]", a, b);
        }

        [DextopRemotableConstructor(route = "rinstant/{routeParam}")]
        public RemoteInstantiationWindow(String routeParam)
        {
            title = "Route with param: " + routeParam;
        }

        [DextopRemotable]
        public RemoteInstantiationWindow(DextopConfig dc)
        {
            title = DextopUtil.Encode(dc);
        }

        public override void InitRemotable(DextopRemote remote, DextopConfig config)
        {
            base.InitRemotable(remote, config);
            config["title"] = title;
        }

        
        
    }
}